Готова 3 практическая работа "Надо сделать идеально"

ESKuchina opened the pull request:
Готова третья практическая работа "Надо сделать идеально". Отправляю на первое ревью в команду Яндекс.Практикум

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Rename fonts.css to font.css

ESKuchina committed:
Update index.html

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Add files via upload

ESKuchina committed:
Update style.css

Avaakum commented on index.html:
```diff
@@ -0,0 +1,34 @@
+    />
+    <meta property="og:url" content="https://practicum.yandex.ru" />
+    <meta property="og:image" content="./images/og.jpg" />
+    <meta property="twitter:card" content="summary_large_image" />
```
---

**Отлично:**

Есть все нужные теги meta

Avaakum commented on index.html:
```diff
@@ -0,0 +1,34 @@
+    <link rel="stylesheet" href="./fonts/font.css" />
+    <link rel="stylesheet" href="./styles/global.css" />
+    <link rel="stylesheet" href="./styles/style.css" />
+  </head>
```
---

**Отлично:**

Корректно заполнен `head`

Avaakum commented on index.html:
```diff
@@ -0,0 +1,34 @@
+      <source src="./video/cover-video.webm" type="video/webm" />
+      <source src="./video/cover-video.mp4" type="video/mp4" />
+    </video>
+    <h1>Надо<br />сделать<br />идеально</h1>
```
Тут теги br не нужны, используйте ограничение максимальной ширины элемента

ESKuchina replied:
Спасибо) Внесла необходимые правки

Avaakum commented on index.html:
```diff
@@ -0,0 +1,34 @@
+  </head>
+  <body>
+    <video autoplay muted loop poster="./images/poster.jpg">
+      <source src="./video/cover-video.webm" type="video/webm" />
```
---

**Отлично:**

Видео в двух форматах

ESKuchina replied:
А я еще думала: заметите или нет?)))) Рада, что заметили) Я старалась)

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Доброго времени суток. У вас получилась хорошая работа!

_Что понравилось:_

- Хорошая структура HTML
- похоже на макет
- Хорошее форматирование документа
- хорошие стили
- используется семантика


_Что необходимо исправить:_

- **Все комментарии, кроме комментариев с пометками «Отлично»**


- Несоответствия макету:
    - https://disk.yandex.ru/i/nbneolsD1f03GQ при ховере заголовок должен быть с прозрачностью помимо плашки




_Что можно улучшить:_


- Селекторы в файле стилей должны располагаться в порядке их появления в html

- В любом месте в файлах CSS в проекте(кроме самого CSS файла подключения шрифтов, где указывается только название подключаемого шрифта), где указано свойство  `font-family`  помимо основного шрифта так же должен быть указан альтернативный шрифт и тип засечек. Это делается для того, что если по каким-то причинами основной шрифт не загрузится, то внешний вид страницы не будет изменен очень сильно.  Порядок написания шрифтов: Основной шрифт, [Веб-безопасный альтернативный шрифт(по ссылке список этих шрифтов)](https://www.cssfontstack.com/), Тип засечек. Пример ` font-family: Inter, Arial, sans-serif; ` 

- Желательно использовать все классы по БЭМ. 
  - https://ru.bem.info/methodology/quick-start/
  - Проверить ошибки БЭМ можно на  https://nglazov.github.io/bem-validator-page/
  - струтуру можно проверить на https://yoksel.github.io/html-tree/

- Можно использовать атрибуты `target="_blank"` для ссылок на сторонние ресурсы для того чтобы пользователю не приходилось заново открывать сайт проекта
- Для более точной корректировки размеров текстов и расстояний между элементами можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi) для Chrome 


**Напоминание:** После выполнения всех критических замечаний работа будет принята. Предложения по улучшению выполняются по желанию.

Желаю успеха!

ESKuchina committed:
Убрала <br> в соответствии с комм-ем от ревьюера

Получила комментарий относительно <h1>: "Тут теги br не нужны, используйте ограничение максимальной ширины элемента", - внесла необходимые правки

ESKuchina committed:
Внесла правки после 1-го ревью

Получила комментарий относительно <h1>: "Тут теги br не нужны, используйте ограничение максимальной ширины элемента", - внесла необходимые правки. В css указала для h1:
max-width: 739px;
margin-left: auto;
margin-right: auto;

Так же изменила последовательность селекторов в файле стилей, чтобы они располагались в порядке их появления в html.

Так же для h1 задала псевдокласс:
h1:hover {
  color: rgba(76, 76, 76, 1);
}
тем самым исправив замечание от ревьюера: "https://disk.yandex.ru/i/nbneolsD1f03GQ при ховере заголовок должен быть с прозрачностью помимо плашки"

ESKuchina committed:
Внесла правки после 1-го ревью

Добавила альтернативный шрифт в body:
font-family: "Raleway", "Arial Black", sans-serif;

ESKuchina committed:
Внесла правки после 1-го ревью

Получила комментарий относительно <h1>: "Тут теги br не нужны, используйте ограничение максимальной ширины элемента", - внесла необходимые правки. В css указала для h1:
max-width: 739px;
margin-left: auto;
margin-right: auto;

Так же изменила последовательность селекторов в файле стилей, чтобы они располагались в порядке их появления в html.

Так же для h1 задала псевдокласс:
h1:hover {
  color: rgba(76, 76, 76, 1);
}
тем самым исправив замечание от ревьюера: "https://disk.yandex.ru/i/nbneolsD1f03GQ при ховере заголовок должен быть с прозрачностью помимо плашки"

Изменила  letter-spacing на 4.06px, чтобы лучше соответствовать макету.

Avaakum reviewed (approved):
Развернуть общий комментарий ⬇︎

Здравствуйте! Рад вам сообщить, что ваша работа принята. Сейчас страничка выглядит прекрасно, всё работает отлично. Отличная семантика и структура. Желаю вам удачи в следующих спринтах и проектах! 



Ссылки:

- Можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`. Но если страница с адаптацией, то всегда стоит проверять как текст с пробелами выглядит на всех ширинах.
- Про альтернативный текст для картинок https://habr.com/ru/post/341810/
- [Ссылка про подгрузку шрифтов](https://only-to-top.ru/blog/tools/2019-03-13-nastrojte-pokaz-vsego-teksta-vo-vremya-zagruzki-veb-shriftov.html) 



Несколько  ссылок на материал по теме нативных CSS переменных от авторов практикума:

- [1 часть](https://tproger.ru/articles/custom-properties-in-css-part-1/)

- [2 часть](https://tproger.ru/articles/custom-properties-in-css-part-2/)

- [3 часть](https://tproger.ru/articles/custom-properties-in-css-part-3/)
